File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if(nums.empty()) return 0;
        unordered_map<int,int> map;
        for(int x = 0; x < nums.size(); x++){
            map[nums[x]]++;
        }
        int ans = 1;
        for(int x = 0; x < nums.size(); x++){
            int pt = nums[x]+1;
            int temp = 1;
            while(map[pt] != 0){
                temp += 1;
                pt++;
            }
            ans = max(temp, ans);
        }
        return ans;
    }
};

File: Data Structures & Algorithms/house-robber/submission-2.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        n = len(nums)
        dp = [-1] * n
        def dfs(i) -> int:
            if i >= n:
                return 0
            if dp[i] != -1:
                return dp[i]
           
            curTotal = max( dfs(i + 1), dfs(i + 2) + nums[i])
            dp[i] = curTotal
            return curTotal

            
        return dfs(0)